### PR TITLE
Tps benchmarks, uds server, stmtparser fix and a whole bunch of other stuff.

### PR DESCRIPTION
**Pull request by feffes**
**OOF**

### CODE REVIEW

**Comment by feffes** [comment]
channel stream isn't being used and can be removed unless you want to keep it @hulthe

**Comment by hulthe** [comment]
Being cross-platform is sort of nice, assuming we can make the performance comparable to UnixStreams. I'll try to look into it tommorow

File: algebraicdb/src/main.rs [hunk 1]
```diff
@@ -19,7 +25,8 @@ struct Config {
 }
 
 #[tokio::main]
-async fn main() -> Result<!, Box<dyn Error>> {
+#[allow(unreachable_code)]
```
Comment (hulthe) [inline algebraicdb/src/main.rs]: What part of this function is unreachable?

File: algebraicdb/src/main.rs [hunk 2]
```diff
@@ -32,5 +39,22 @@ async fn main() -> Result<!, Box<dyn Error>> {
...
+            .unwrap()
+    });
+    tokio::spawn(async move {
+        create_uds_server(PathBuf::from(uds_address), uds_state)
+            .await
+            .unwrap()
+    });
+
+    ctrl_c().await.unwrap();
+    info!("Shutting down...");
```
Comment (hulthe) [inline algebraicdb/src/main.rs]:
```suggestion
    info!("shutting down...");
```
let's be consistent

File: algebraicdb/src/state/dbms.rs [hunk 3]
```diff
@@ -111,8 +111,12 @@ impl DbmsState {
             let db_data = match load_db_data(&config.data_dir).await {
                 Ok(state) => state,
                 Err(e) => {
-                    info!("Failed to read stored data from disk, is this a fresh instance? {}", e);
-                    initialize_data_dir(&config.data_dir).await
+                    info!(
+                        "Failed to read stored data from disk, is this a fresh instance? {}",
```
Comment (hulthe) [inline algebraicdb/src/state/dbms.rs]:
```suggestion
                        "failed to read stored data from disk, is this a fresh instance? {}",
```

File: benches/pgbench_benchmark.rs [hunk 4]
```diff
@@ -0,0 +1,138 @@
+#![feature(async_closure)]
```
Comment (hulthe) [inline benches/pgbench_benchmark.rs]: Doesn't look like this feature is used
```suggestion
```

File: benches/src/runtime.rs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+pub fn rt() -> Runtime {
+    runtime::Builder::new()
+        .threaded_scheduler()
+        .core_threads(10)
+        .thread_name("normal-runtime")
+        .enable_all()
+        .build()
+        .unwrap()
+}
```
Comment (hulthe) [inline benches/src/runtime.rs:30]: I feel like these functions could just be one function with a name argument.
```rust
pub fn create_runtime(name: &str) -> Runtime { ... }
```

File: benches/src/tps/mod.rs [hunk 6]
```diff
@@ -1 +1,3 @@
-//TODO
\ No newline at end of file
+//TODO
```
Comment (hulthe) [inline benches/src/tps/mod.rs]: What is there left to do?

**Review by hulthe — APPROVED** [review]

**Pull request merged**